include/tc/lang/lexer.h

```diff
@@ -371,9 +382,20 @@ struct Lexer {
     next();
     return true;
   }
+  Token lookahead() {
+    if (!lookahead_) {
+      lookahead_.reset(new Token(lex()));
+    }
+    return *lookahead_;
```

Comment:
`lookahead()` and `cur()` need to return the same thing, and there is no concern about deleteion because the both return by value so the pointer to lookahead never leaks from the lexer. The unique_ptr is just used internally to know whether `lookahead()` has been defined.